### PR TITLE
@orta -> snapshots can automatically be named after the spec

### PR DESCRIPTION
**Pull request by 1aurabrown**
I gave this a shot with the show view controller, since that's what sarah and i had been looking at.

This makes it possible to put the device snapshot macro inside a shared examples block, which wasn't possible before because every time the shared example was run the same image file would be re-written.

for future reference `sanitizedTestPath()` is the very useful method used to get the current test name. I had an implementation that used this, but it turned out to be unnecessary as it's handled by fb snapshots' methods.

### CODE REVIEW

File: Artsy Tests/ARShowViewControllerTests.m [hunk 1]
```diff
@@ -13,14 +13,73 @@ - (void)addMapPreview;
...
+
+                [showVC ar_presentWithFrame:[[UIScreen mainScreen] bounds]];
+                [showVC.view snapshotViewAfterScreenUpdates:YES];
+
+                return showVC;
+            });
+        });
+
+        describe(@"with 1 install image", ^{
+            itHasSnapshotsForDevices(^{
```
Comment (orta) [inline Artsy Tests/ARShowViewControllerTests.m]: elegant.

**Comment by orta** [comment]
I really like how this has turned out :+1:

**Comment by 1aurabrown** [comment]
alkdjflaskjfd i tried to rebase and made a bad thing happen

**Comment by orta** [comment]
hah, yep. maybe try squash your three commits into one ( and give it a good name? )

**Comment by 1aurabrown** [comment]
@orta i squashed

**Comment by orta** [comment]
:european_castle:

**Pull request merged**